### DISCUSSION TITLE
Relation usage

### DISCUSSION THREAD

**Discussion by NuclleaR** [strapi/strapi#5955] [Legacy Topics]
Hello!
I'm using the latest version of Strapi with PostgreSQL 12.

I tried to play with relations between models. And what I've found that confusing me, that I did not find any foreign key relationships between tables. I created the model `Category` and model `Product` with relation to category many to one. When I looked through the database I found that the Product has a `category` column with just type integer and no foreign keys at all. I tried to change integer for any non-existing category ID and it saved successfully. Maybe I did some mistake but when I add relations I expect to use foreign keys between tables. Could someone help me with the relation model?

Many thanks for any help 🙏

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions)

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Reply by NuclleaR** [reply]
Sorry, didn't see discussion section. Thx

**Comment by alexandrebodin** [comment]
Hi @NuclleaR, Thank your interest :) 

For legacy reasons we haven't yet added foreign keys. Right now strapi tries to avoids running database migrations on startup as much as possible to avoid messing with the data. 

In order to support foreign keys we will need to handled cascading / column renaming and such. Which is a lot harder to do dynamically than when you are building it yourself with migration files. To do so we have a lot of work to do tbh :) This is mainly why it is not implemented yet but we really want to improve this in the future.

**Reply by NuclleaR** [reply]
thx
